lms/djangoapps/verify_student/views.py

```diff
@@ -305,7 +317,9 @@ def get(
         # with a paid course mode (such as "verified").
         # For this reason, every paid user is enrolled, but not
         # every enrolled user is paid.
-        already_verified = self._check_already_verified(request.user)
+        # If the course mode is not verified(i.e only paid) then already_verified is always True
+        already_verified = self._check_already_verified(request.user) \
+            if CourseMode.is_verified_mode(course_mode) else True
```

Comment:
It looks like you're also doing this check in `_display_steps()` -- maybe it makes sense for `_display_steps()` to check if the user is already verified instead of doing it here.